### ISSUE TITLE
Helm uninstall of application doesn't remove synced K8s secrets

### ISSUE THREAD

**Issue by SatyKrish** [kubernetes-sigs/secrets-store-csi-driver#292]
**What steps did you take and what happened:**
Application is configured to sync secrets from Azure Key Vault with K8s secrets, and use secrets as Environment Variables. Application deployed using helm chart. 

Ran following tests:

1. Delete deployment referencing SecretProviderClass manually.
[Success] - Pods get deleted; synced K8s secrets get deleted.

```
$ kubectl delete deploy secrets-provider -n test
deployment.apps "secrets-provider" deleted
```

```
$ kubectl get secrets -n test
NAME                                   TYPE                                                      DATA   AGE
default-token-xzn9w            kubernetes.io/service-account-token   3         8h
secrets-provider-credentials   Opaque                                              2         8h
```

2. Uninstall helm release for the application.
[Failed] - Pods get deleted; synced secrets are still found in K8s secrets.

```
$ helm uninstall test -n test
release "test" uninstalled
```

```
$ kubectl get secrets -n test
NAME                                                 TYPE                                                      DATA   AGE
secret/default-token-xzn9w               kubernetes.io/service-account-token   3         8h
secret/ingress-tls-certificate               kubernetes.io/tls                                  2          4h44m
secret/secrets-provider-credentials   Opaque                                                2          8h
```

**What did you expect to happen:**
After helm uninstall, secrets synced with K8s secrets should be deleted.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Which provider are you using:**
Azure Key Vault

**Environment:**

- Secrets Store CSI Driver version: (use the image tag): v0.0.11
- Kubernetes version: (use `kubectl version`): 1.16.10

**Comment by aramase** [comment]
@SatyKrish Thank you for opening the issue.

Did you get a chance to try out our lates available release [v0.0.12](https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v0.0.12)? We have switched to using a controller for creating secret and the pod that the secrets are created for are added as owners `ownerReferences`. So when all pods using the secret are deleted, then the secret will automatically be garbage collected. 

Try out `v0.0.12` and let us know if you're still having issues.

**Comment by aramase** [comment]
Closing this issue. Please feel free to reopen if you're having issues with the latest releases.

/close
